### DISCUSSION TITLE
How can I transfer my PalmNote data export file to Standard Notes?

### DISCUSSION THREAD

**Discussion by c182** [standardnotes/forum#2760] [Suggestions]
**Describe the issue**
The Android PalmNote app has been abandoned by Google Play. I would like to transfer my note data records in my PalmNote app on my old smartphone to Standard Notes on my new smartphone.   PalmNote will export my note data records to an XML file. Is there a way to import the PalmNote XML file into Standard Notes??  Please recommend a process to accomplish this.
.
**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser:  Chrome, Brave, FireFox
 - Version:  The latest

**Smartphone (please complete the following information):**
 - Device: Motorola Edge (2021)
 - OS: Android 11
 - Browser:  Brave, Chrome
 - Version: The latest

**Additional context**
The obsolete Android PalmNote app will only export its data records to an XML file (no other format is available)

**Comment by JaspalSuri** [comment]
Hi @c182, I'm afraid that we currently do not have a tool for converting notes exported from the Palm Notes app into a file that is usable by the Standard Notes apps. 

Can you provide us with an example of how the exported file lists a single note entry? Make sure to remove any private or otherwise sensitive information first. You should be able to open the XML file within any text editor (unless the notes contain images).

**Comment by c182** [comment]
Hello Jaspal,

I have attached a sample of the XML file exported by PalmNote.
I had to change to file extension from XML to TXT in order to attach it to this reply.
The file begins with a PalmNote header followed by three (3) data records.
The third data record shows how the export file ended.

I'm not a programmer. Here is my take on what I see.
Start of each data record appears to begin with the text:

`[CDATA[`

Each data record appears to end with the text:

`]]>`

PalmNote internal "housekeeping" text for each data record starts at the end of each data record and ends at the start of the next data record

The above text is not visible to the Palmnote user

I hope this helps. 
Let me know if you have any questions.

Respectfully,
C1182

```
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<!--PalmNote xml export-->
<notes version="1.0">
	<note>
		<category>Palm</category>
		<text>
			<![CDATA[NOTE BODY 1]]>
		</text>
		<created>1370671781810</created>
		<modified>1410410336385</modified>
		<readonly>0</readonly>
	</note>
	<note>
		<category>Palm</category>
		<text>
			<![CDATA[NOTE BODY 2]]>
		</text>
		<created>1370671781821</created>
		<modified>1370671781821</modified>
		<readonly>0</readonly>
	</note>
	<note>
		<category>Palm</category>
		<text>
			<![CDATA[NOTE BODY 3]]>	
		</text>
		<created>1625602634538</created>
		<modified>1625602806315</modified>
		<readonly>0</readonly>
	</note>
</notes>
```

**Comment by JaspalSuri** [comment]
Hi @c182,

Thanks for the attachment. While we generally don't do this given the various formats used by different note taking apps, I'll ask the dev team if it would be possible to write a script that will convert the XML file into a backup that Standard Notes can use.

[According to this website](https://www.epochconverter.com/), two of the notes were created and modified in 2013, while one was created and modified in 2021. Assuming you have access to the PalmNote application, does that appear to be correct?

**Comment by c182** [comment]
Hello Jaspal Suri:
I opened the PalmNote app on my old cell phone and verified you are correct that two of the notes were created and modified in 2013, while one was created and modified in 2021.
I am very excited to hear that you will ask the dev team if it would be possible to write a script that will convert the XML file into a backup that Standard Notes can use.  I very much hope it will be possible.  I hope to use Standard Notes on all my devices. Thank you so much!! 
C182

**Comment by JaspalSuri** [comment]
Hello @c182, you're very welcome! Thank you for taking the time to confirm that! Please make sure that your exported backup is stored somewhere safe. It might be a while from now, but I will let you know when I hear back from the dev team on this. In the meantime you can use Standard Notes and import the backup later. 🙂
